crates/cdk/src/wallet/mod.rs

```diff
@@ -336,8 +336,15 @@ impl Wallet {
                     })
                     .collect::<Result<Vec<ProofInfo>, _>>()?;
 
+                let spent_public_keys: Vec<PublicKey> = proofs
+                    .iter()
+                    .zip(states)
+                    .filter(|(_, state)| state.state == State::Spent)
+                    .filter_map(|(p, _)| p.y().ok())
+                    .collect();
+
```

Comment:
```suggestion
        let spent_ys: Vec<_> = spendable
            .states
            .iter()
            .filter_map(|p| match p.state {
                State::Spent => Some(p.y),
                _ => None,
            })
            .collect();
```

This can be done in one filtermap. If the state is not spent we do not want update the state with the mint state. This is because the mint does not know about the reserved state so it will say a proof is unspent when it should be reserved.

Reply:
Alright, if the mint does not know about reserved state, removing only the spent state is the right option.

Reply:
@thesimplekid where is the `spendable ` vec coming from